jdbc/src/main/java/org/apache/metamodel/jdbc/JdbcDataContext.java

```diff
@@ -778,18 +780,33 @@ private String findDefaultSchema(final String defaultName, final List<String> sc
      * @throws SQLException
      */
     private Set<String> getSchemaSQLServerNames(DatabaseMetaData metaData) throws SQLException {
-        // Distinct schema names. metaData.getTables() is a denormalized
-        // resultset
-        Set<String> schemas = new HashSet<>();
-        ResultSet rs = null;
+        // Distinct schema names. metaData.getTables() is a denormalized resultset
+        final Set<String> schemas = new HashSet<>();
+        ResultSet resultSet = null;
+
+        // Find DBO schema if present
         try {
```

Comment:
Since `ResultSet` is `AutoCloseable`, isn't it better to change this to try-with-resources?

Reply:
I wanted to implement that, but then the FileHelper#safeClose(Object... objects) won't be called anymore. I'm not sure if this is a better method then what's implemented by the AutoCloseable#close() implementation?

Reply:
I personally think we should use the try-with-resources. The `FileHelper#safeClose(Object)` is most probably used right now because this functionality was originally implemented in a version of java which didn't support the try-with-resources construct yet. @kaspersorensen Am I correct regarding this?

Reply:
Well yes and no. There is a difference... the `safeClose` method will log but ignore any exceptions. It was built from a mentality of "close this resource, but if that fails I don't know that we can do anything about it".

I'm OK with changing it though.

Reply:
Implemented try-with-resources